### PR TITLE
SOF: Update uapi/sof-abi.h header to conform to current SOF master

### PR DESCRIPTION
**Pull request by akloniex**
Signed-off-by: ArturX Kloniecki <arturx.kloniecki@linux.intel.com>

### CODE REVIEW

**Comment by ranj063** [comment]
@lgirdwood this patch is required for the eq pipelines to work. Could we please merge it?

**Comment by keyonjie** [comment]
@akloniex can you fix the checkpatch errors please?

**Comment by lgirdwood** [comment]
ping @akloniex - can you fix the check patch errors

File: include/uapi/sound/sof-abi.h [hunk 1]
```diff
@@ -9,7 +9,23 @@
...
+#define SOF_ABI_VER(major, minor, micro) \
+	(((major)<<8)|((minor)<<4)|(micro))
+#define SOF_ABI_VERSION_MAJOR(version)	(((version)>>8) & 0xff)
+#define SOF_ABI_VERSION_MINOR(version)	(((version)>>4) & 0xf)
+#define SOF_ABI_VERSION_MICRO(version)	((version) & 0xf)
+#define SOF_ABI_VERSION_INCOMPATIBLE(sof_ver, client_ver) \
+	(SOF_ABI_VERSION_MAJOR(sof_ver) != SOF_ABI_VERSION_MAJOR(client_ver) ||\
+	(SOF_ABI_VERSION_MAJOR(sof_ver) == SOF_ABI_VERSION_MAJOR(client_ver) &&\
+	SOF_ABI_VERSION_MINOR(sof_ver) != SOF_ABI_VERSION_MINOR(client_ver)))
+
```
Comment (plbossart) [inline include/uapi/sound/sof-abi.h:27]: always use parentheses for macro arguments.

Comment (akloniex) [reply include/uapi/sound/sof-abi.h:27]: done

**Comment by akloniex** [comment]
@lgirdwood @plbossart 
Sorry for the delay in response to this.
I've fixed most of checkpatch issues, only macro argument reuse is reported now. I don't think that it's fixable or needs any actual adjustments.

**Comment by plbossart** [comment]
yes, it's fine, but you need to apply the same change on sof as well. We want the files to be completely identical on both sides. can you add a PR for sof and i'll merge both sof and kernel sides at once. Thanks!

**Comment by akloniex** [comment]
@plbossart I've already sent PR to sof repository. https://github.com/thesofproject/sof/pull/502

**Pull request merged**